sway-lib-std/src/identity.sw

```diff
@@ -21,3 +24,62 @@ impl core::ops::Eq for Identity {
         }
     }
 }
+
+impl Identity {
+    pub fn as_address(self) -> Option<Address> {
+        match self {
+            Identity::Address(address) => Option::Some(address),
+            Identity::ContractId(_) => Option::None(),
+        }
+    }
+
+    pub fn as_contract_id(self) -> Option<ContractId> {
+        match self {
+            Identity::Address(_) => Option::None(),
+            Identity::ContractId(contract_id) => Option::Some(contract_id),
+        }
+    }
+
+    pub fn is_address(self) -> bool {
+        match self {
+            Identity::Address(_) => true,
+            Identity::ContractId(_) => false,
+        }
+    }
+
+    pub fn is_contract_id(self) -> bool {
+        match self {
+            Identity::Address(_) => false,
+            Identity::ContractId(_) => true,
+        }
+    }
+
+    pub fn value(self) -> b256 {
+        match self {
+            Identity::Address(address) => address.value,
+            Identity::ContractId(contract_id) => contract_id.value,
+        }
+    }
```

Comment:
```suggestion
```

This should not be added because it defeats the purpose of using the wrapped types, doubly so on the `Identity`.

Reply:
I disagree, you can use the wrapper and then still the actual value for some other stuff, like low level instruction calls, this is convenient for that

Reply:
"I disagree" is not a valid counter argument to type safety.

This does not belong on the `Identity` type. Users can extract it themselves via the wrapped types.

Reply:
Its not breaking type safety, you can just not use this function. Users can already access the values yes, and this is just a utility method to make that easier. Address and ContractId also have a into() fn for converting to b256 so I dont see why this is any worse than that

Reply:
The value is already a private type that at best should only be accessible through the wrapper if it allows it. The `Identity` is another wrapper on top of that, a super type. It should have no concept of what its variants' internal structure is like. That data is encapsulated for a reason. If the user wants the internal value then it needs to query it via the wrapper.

@IGI-111 looking at the field, it is a private field but we do not respect that. Is that intentional or should we raise an issue and force the use of the `pub` keyword on the fields?

Reply:
We don't have privacy for struct fields yet. Though that should be fixed.
In any case I don't think we should be writing code into the standard library that assumes fields are public like this.

But the real reason not to do this is that it is a type system escape hatch. This methods allows you to mix the two types of identities without explicitly ascertaining which one you are using, and that's bad design. I vote for removal too. If you want the raw value of something, you should know what it is first. What use is the value otherwise?